pkg/services/annotations/annotationsimpl/xorm_store.go

```diff
@@ -378,3 +379,22 @@ func (r *SQLAnnotationRepo) GetTags(ctx context.Context, query *annotations.Tags
 
 	return annotations.FindTagsResult{Tags: tags}, nil
 }
+
+func (r *SQLAnnotationRepo) validateItem(item *annotations.Item) error {
+	if err := validateTimeRange(item); err != nil {
+		return err
+	}
+
+	if err := r.validateTagsLength(item); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *SQLAnnotationRepo) validateTagsLength(item *annotations.Item) error {
+	tagsStr := fmt.Sprintf("%v", item.Tags)
```

Comment:
This is definitely something that shouldn't affect merging this, as the only consequence of getting this wrong would be certain border values getting a less helpful error message.

I think [Xorm's type conversion](https://gitea.com/xorm/xorm/src/tag/v0.8.2/session_convert.go#L643-L649) uses json.Marshal here, so it's possible that this validation will pass but the later validation fail because of the difference between Go's `[a b c]` and JSON's `["a", "b", "c"]` 🤔

Presumably, another way of getting an estimate of the length without having to convert it to JSON would be something like:

```
estimatedTagLength := len(item.Tags) * 4 // each item has approximately a pair of quotes, a comma and a space for each object except for the last, plus two brackets.
for _, t := range item.Tags {
	estimatedTagLength += len(t)
}
```

Reply:
good point, I will fix it